datalad/distribution/dataset.py

```diff
@@ -204,6 +206,47 @@ def get_subdatasets(self, pattern=None, fulfilled=None, absolute=False,
         else:
             return submodules
 
+    def create_subdataset(self, path, name=None,
+                          description=None,
+                          no_annex=False,
+                          annex_version=None,
+                          annex_backend='MD5E',
+                          git_opts=None,
+                          annex_opts=None,
+                          annex_init_opts=None):
+
+        # TODO: use `name` for subdatasets. (not necessarily equals `path`)
+        # TODO: check whether path is in self?
+        subds = Dataset(path)
+
+        # create the dataset
+        subds.create(description=description,
+                     no_annex=no_annex,
+                     annex_version=annex_version,
+                     annex_backend=annex_backend,
+                     git_opts=git_opts,
+                     annex_opts=annex_opts,
+                     annex_init_opts=annex_init_opts,
+                     # Note:
+                     # adding to the superdataset is what we are doing herein!
+                     # add_to_super=True would lead to calling ourselves again
+                     # and again
+                     # While this is somewhat ugly, the issue behind this is a
+                     # necessarily slightly different logic of `create` in
+                     # comparison to other toplevel functions, which operate on
+                     # an existing dataset and possibly on subdatasets.
+                     # With `create` we suddenly need to operate on a
+                     # superdataset, if add_to_super is True.
+                     add_to_super=False)
+
+        # add it as a submodule
+        # TODO: clean that part and move it in here (Dataset)
+        #       or call install to add the thing inplace
+        from .install import _install_subds_inplace
```

Comment:
Well, it's basically installing a dataset, I think. If you create a dataset within another one without `add_to_super` and you want to add it later on, this should then be done by `install`or `add`. So, I would prefer to wait for these to be ready and call the correct one from here. Agree?